### PR TITLE
(FACT-2406) Fix custom and external facts loading.

### PR DESCRIPTION
**Pull request by BogdanIrimie**

### CODE REVIEW

File: lib/custom_facts/util/fact.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Fact
       def initialize(name, options = {})
         @name = name.to_s.downcase.intern
 
+        @options = ::Facter::Utils.deep_copy(options)
```
Comment (gimmyxd) [inline lib/custom_facts/util/fact.rb]: This should work evenif Facter is not globally namespaced: `Facter::Utils.deep_copy(@options.merge(options))`

File: lib/custom_facts/util/fact.rb [hunk 2]
```diff
@@ -52,7 +53,7 @@ def initialize(name, options = {})
       #
       # @api private
       def add(options = {}, &block)
-        @options = options
+        @options = ::Facter::Utils.deep_copy(@options.merge(options))
```
Comment (gimmyxd) [inline lib/custom_facts/util/fact.rb]: This should work evenif Facter is not globally namespaced: `Facter::Utils.deep_copy(@options.merge(options))`

File: lib/custom_facts/util/fact.rb (hunk 1 above)
Comment (IrimieBogdan) [reply lib/custom_facts/util/fact.rb]: Removed global namespace.

File: lib/custom_facts/util/fact.rb (hunk 2 above)
Comment (IrimieBogdan) [reply lib/custom_facts/util/fact.rb]: Removed global namespace.

**Review by gimmyxd — APPROVED** [review]

**Pull request merged**